CalibTracker/SiStripESProducers/plugins/fake/SiStripHashedDetIdFakeESSource.cc

```diff
@@ -1,70 +1,52 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/TrackerNumberingBuilder/interface/GeometricDet.h"
+#include "Geometry/TrackerNumberingBuilder/interface/utils.h"
+
 #include "CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h"
 #include "CalibTracker/Records/interface/SiStripHashedDetIdRcd.h"
-#include "CalibTracker/SiStripCommon/interface/SiStripDetInfoFileReader.h"
-#include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include <sstream>
-#include <vector>
-#include <map>
+
+class SiStripHashedDetIdFakeESSource : public edm::ESProducer {
+public:
+  explicit SiStripHashedDetIdFakeESSource(const edm::ParameterSet&);
+  ~SiStripHashedDetIdFakeESSource() override;
+
+  virtual std::unique_ptr<SiStripHashedDetId> produce(const SiStripHashedDetIdRcd&);
+
+private:
+  edm::ESGetToken<GeometricDet, IdealGeometryRecord> geomDetToken_;
+};
 
 using namespace sistrip;
 
-// -----------------------------------------------------------------------------
-//
-SiStripHashedDetIdFakeESSource::SiStripHashedDetIdFakeESSource( const edm::ParameterSet& pset )
-  : SiStripHashedDetIdESProducer( pset )
-{
-  findingRecord<SiStripHashedDetIdRcd>();
-  edm::LogVerbatim("HashedDetId") 
-    << "[SiStripHashedDetIdFakeESSource::" << __func__ << "]"
-    << " Constructing object...";
+SiStripHashedDetIdFakeESSource::SiStripHashedDetIdFakeESSource(const edm::ParameterSet& pset) {
+  setWhatProduced(this).setConsumes(geomDetToken_);
 }
 
-// -----------------------------------------------------------------------------
-//
-SiStripHashedDetIdFakeESSource::~SiStripHashedDetIdFakeESSource() {
-  edm::LogVerbatim("HashedDetId")
-    << "[SiStripHashedDetIdFakeESSource::" << __func__ << "]"
-    << " Destructing object...";
-}
+SiStripHashedDetIdFakeESSource::~SiStripHashedDetIdFakeESSource() {}
+
+std::unique_ptr<SiStripHashedDetId> SiStripHashedDetIdFakeESSource::produce(const SiStripHashedDetIdRcd& record) {
+  edm::LogVerbatim("HashedDetId") << "[SiStripHashedDetIdFakeESSource::" << __func__ << "]"
+                                  << " Building \"fake\" hashed DetId map from IdealGeometry";
 
-// -----------------------------------------------------------------------------
-// 
-SiStripHashedDetId* SiStripHashedDetIdFakeESSource::make( const SiStripHashedDetIdRcd& ) {
-  edm::LogVerbatim("HashedDetId")
-    << "[SiStripHashedDetIdFakeESSource::" << __func__ << "]"
-    << " Building \"fake\" hashed DetId map from ascii file";
-  
-  typedef std::map<uint32_t,SiStripDetInfoFileReader::DetInfo>  Dets;
-  const edm::Service<SiStripDetInfoFileReader> reader;
-  Dets det_info = reader->getAllData();
-  
-  std::vector<uint32_t> dets;
-  dets.reserve(16000);
+  const auto& geomDet =
+      record.getRecord<TrackerDigiGeometryRecord>().getRecord<IdealGeometryRecord>().get(geomDetToken_);
```

Comment:
This could be shortened to 
```suggestion
      record.getRecord<TrackerDigiGeometryRecord>().get(geomDetToken_);
```